iree/hal/cuda/dynamic_symbols.c

```diff
@@ -20,6 +20,14 @@ static const char* kCUDALoaderSearchNames[] = {
 #endif
 };
 
+static const char* kCUPTILoaderSearchNames[] = {
```

Comment:
in the future we will want to support searching the `CUDA_PATH` environment variable in cases where cuda/cupti is not on the system path or the user wants to use a different cuda - if we can get away with this for now that's fine

Reply:
This works on my machine after quite a bit of effort installing cupti and messing around with several environment variables to get the dynamic library loader to find it on Debian. Can't comment on how this will work on other platforms, particularly Windows.

Reply:
A few of us are also chatting offline. I think https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html#cupti can help with library discovery here.

Reply:
@ScottTodd @GMNGeoffrey  Trying `FindCUDAToolkit` didn't work on its own. The package produces build targets, not header files. Even more problematic, when inspecting the targets for CUPTI, I get this information on my Debian machine: 

```
[cmake]  Properties for TARGET CUDA::cupti:
[cmake]    CUDA::cupti.LOCATION = "/usr/local/cuda/extras/CUPTI/lib64/libcupti.so"
[cmake]    CUDA::cupti.INCLUDE_DIRECTORIES = <NOTFOUND>
[cmake]    CUDA::cupti.INSTALL_RPATH = <NOTFOUND>
[cmake]    CUDA::cupti.INTERFACE_LINK_DIRECTORIES = <NOTFOUND>
[cmake]    CUDA::cupti.INTERFACE_SYSTEM_INCLUDE_DIRECTORIES = "/usr/local/cuda/include"
[cmake]    CUDA::cupti.CUDA_RUNTIME_LIBRARY = <NOTFOUND>
[cmake]    CUDA::cupti.INTERFACE_INCLUDE_DIRECTORIES = "/usr/local/cuda/include"
```

The properties point to the base CUDA includes, which are different from the correct CUPTI values.

I found a more involved work around, that involves using `find_package` to locate the header file directories from the CUDA install. This works but it has some manual directory paths that may be brittle. Would help if someone could check this on Windows? 

If this checks, we can delete all of the source in `third_party/cuda/include` and just rely on the user's installed CUDA header files. I don't believe this will affect builds in practice, since you still need an actual CUDA install for linking anyway.